src/cloudai/workloads/megatron_bridge/megatron_bridge.py

```
@@ -575,8 +577,36 @@ def was_run_successful(self, tr: TestRun) -> JobStatusResult:
+            str(tr.output_path), "experiments", experiment_name, "*", experiment_name, "torch_profile"
+        )
+        matches = [p for p in glob.glob(pattern) if os.path.isdir(p)]
+        if not matches:
+            return
+
+        # Take the most recently modified in case of retries
+        src = max(matches, key=os.path.getmtime)
+        dst = tr.output_path / "torch_profile"
+        if not dst.exists():
+            shutil.copytree(src, dst)
+            logging.info(f"Copied pytorch profiler traces from {src} to {dst}")
```

Ruff (0.16.0)

[warning] Logging statement uses f-string

(G004)